### ISSUE TITLE
Front/Back issues with a HwndHost

### ISSUE THREAD

**Issue by smourier** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1321]
I'm using a custom HwndHost with the MaterialDesignInXamlToolkit. I've basically implemented what's described officially here: https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/walkthrough-hosting-a-win32-control-in-wpf

But it poses two problems

1) problem 1: when I open a Dialog, the HwndHost is not dimmed, see here:

![matxaml1](https://user-images.githubusercontent.com/5328574/61175405-ff710400-a5ae-11e9-8faf-e0987699b590.png)

2) problem 2: when I open a Drawer, the HwndHost stays on top of the drawer, see here:

![matxaml2](https://user-images.githubusercontent.com/5328574/61175407-04ce4e80-a5af-11e9-9d2f-2597e23492f3.png)

This is the window XAML code (no code behind):

```
<Window x:Class="WpfApp2.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:local="clr-namespace:WpfApp2"
        xmlns:md="http://materialdesigninxaml.net/winfx/xaml/themes"
        Title="Test" Width="600" Height="600">
    <Grid>
        <md:DialogHost>
            <md:DialogHost.DialogContent>
                <StackPanel Margin="10">
                    <TextBlock Margin="10">This is a dialog box</TextBlock>
                    <Button Command="{x:Static md:DialogHost.CloseDialogCommand}">Close</Button>
                </StackPanel>
            </md:DialogHost.DialogContent>
            <md:DrawerHost IsLeftDrawerOpen="{Binding ElementName=MenuToggleButton, Path=IsChecked}">
                <md:DrawerHost.LeftDrawerContent>
                    <Border Background="Blue" Width="50">
                        <TextBlock Foreground="White">Drawer</TextBlock>
                    </Border>
                </md:DrawerHost.LeftDrawerContent>
                <StackPanel>
                    <ToggleButton Style="{StaticResource MaterialDesignHamburgerToggleButton}" IsChecked="False" x:Name="MenuToggleButton" />
                    <Button Command="{x:Static md:DialogHost.OpenDialogCommand}">Open Dialog</Button>
                    <Border Background="Red" Height="200"/>
                    <local:HostWindow Height="200" />
                </StackPanel>
            </md:DrawerHost>
        </md:DialogHost>
    </Grid>
</Window>

```
This is the HWndHost pretty standard code:

    public class HostWindow : HwndHost
    {
        protected override HandleRef BuildWindowCore(HandleRef hwndParent)
        {
            const int WS_VISIBLE = 0x10000000;
            const int WS_CHILD = 0x40000000;
            var hwnd = CreateWindowEx(0, "static", string.Empty, WS_CHILD | WS_VISIBLE, 0, 0, 0, 0, hwndParent.Handle, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
            return new HandleRef(this, hwnd);
        }

        protected override void DestroyWindowCore(HandleRef hwnd) => DestroyWindow(hwnd.Handle);

        protected override IntPtr WndProc(IntPtr hwnd, int msg, IntPtr wParam, IntPtr lParam, ref bool handled)
        {
            handled = false;
            return IntPtr.Zero;
        }

        [DllImport("user32")]
        private static extern bool DestroyWindow(IntPtr hwnd);

        [DllImport("user32", CharSet = CharSet.Unicode)]
        private static extern IntPtr CreateWindowEx(int dwExStyle, string lpClassName, string lpWindowName, int dwStyle, int X, int Y, int nWidth, int nHeight, IntPtr hwndParent, IntPtr hMenu, IntPtr hInstance, IntPtr lpParam);
    }

I don't know how can I fix this, I'm open to any idea?

**Comment by jespersh** [comment]
Sorry, but there is nothing you can do:
https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/layout-considerations-for-the-windowsformshost-element#z-order

**Comment by smourier** [comment]
Hi,

Thanks for your answer.

I actually spend the whole weekend trying to find a solution and I did :-)

I'm not using the `WinformsHostElement` but the problem is the same.  It's not MaterialDesignInXamlToolkit-related and is know as the WPF "airspace" issue:  https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-3.0/aa970688(v=vs.85) and https://docs.microsoft.com/en-us/dotnet/framework/wpf/advanced/technology-regions-overview

In my case, I needed a Windows' native HWND because I wanted to have a DirectX11 surface  used as a control in my App.

WPF has the `D3DImage` control but sadly this only supports DirectX9 surfaces, not DirectX11 ones.

So I finally found this `D3D11Image` sample project from Microsoft: https://github.com/microsoft/WPFDXInterop and now it works!
